### DISCUSSION TITLE
How to send the file from S3 to Client with response ?

### DISCUSSION THREAD

**Discussion by bumjun-kim** [aws/aws-sdk-js-v3#3838] [Q&A]
I am trying to make a download API for client based on TypeScript.

Object can be read from S3, but cannot send it to client with response for some reason.

Can anyone find what I made wrong? 

Below is my code...

```js
    downloadFile = async (req: Request, res: Response): Promise<void> => {
      const fileName = 'folder/3c96c1b';

      const getObjectCommand = new GetObjectCommand({ Bucket: config.awsS3Bucket, Key: fileName });
      const result = await awsS3.send(getObjectCommand);

      if (result.Body) {
        const stream = result.Body as Readable;

        console.log(111, Object.keys(result.Body));

        stream.pipe(res);
      }
    };
```

**Comment by aBurmeseDev** [comment]
Hi @bumjun-kim - thanks for reaching out. 

>Object can be read from S3, but cannot send it to client with response for some reason.

I'm not sure what you mean by cannot send it to client with response. I don't see anything wrong with the code you have on GetObjectCommand. Are you getting any errors?
